drivers/ethernet/Kconfig.nxp_imx_netc

```diff
@@ -7,6 +7,7 @@ menuconfig ETH_NXP_IMX_NETC
 	depends on DT_HAS_NXP_IMX_NETC_PSI_ENABLED
 	select MDIO
 	select NOCACHE_MEMORY if ARCH_HAS_NOCACHE_MEMORY_SUPPORT
+	select ETH_DSA_SUPPORT
```

Comment:
Should this driver Kconfig select DSA support, or should the `DSA_NXP_IMX_NETC` Kconfig select it?

Reply:
Good question. I think it makes sense to rework this.

```
config ETH_DSA_SUPPORT
        bool
        help
          Set by an ethernet driver that supports DSA.

menuconfig NET_DSA
        bool "Distributed Switch Architecture support"
        depends on ETH_DSA_SUPPORT
        help
          Enable Distributed Switch Architecture support. For now it
          only supports Kinetics and STM32 ENET drivers.
```

NET_DSA should select ETH_DSA_SUPPORT.  Then this is handle in common NET option, and device driver no longer needs to handle. Also it's possible eth device driver supports DSA master mode without any code wrappered with ETH_DSA_SUPPORT.

```
./drivers/ethernet/Kconfig.sam_gmac:    select ETH_DSA_SUPPORT
./drivers/ethernet/nxp_enet/Kconfig:    select ETH_DSA_SUPPORT
./drivers/ethernet/nxp_enet/Kconfig:    select ETH_DSA_SUPPORT
./drivers/ethernet/Kconfig.stm32_hal:   select ETH_DSA_SUPPORT
```

Thanks.